### DISCUSSION TITLE
[Question] Can I define a new method notification?

### DISCUSSION THREAD

**Discussion by nvlong198** [apache/superset#16004] [Q&A / Help] [answered]
Currently, we can alert & report send to email/slack, but I want them to send to my 3rd app or simply save a snapshots' dashboard to the file system.
Is there any way to config a new method of notification and action for each method?
![image](https://user-images.githubusercontent.com/23328191/127774857-484b21ae-a8b8-4b69-b578-c5542cf96f16.png)

**Comment by nytai** [comment]
You would need to change source code to support this

**Reply by nytai** [reply, marked as answer]
You just need to make changes in this module https://github.com/apache/superset/tree/911462a148145e886f0553c18f0b643c8f69cf68/superset/reports/notifications

**Comment by abifauzann** [comment]
Hi @nvlong198 , how did you enable Slack in your dropdown notification method ?
Mine has followed Alert and Report guide from the official docs but no Slack method, only email. 
<img width="1181" alt="image" src="https://github.com/apache/superset/assets/110159065/fa01bb5c-4609-4e1b-b19a-7abd90004ede">

**Comment by Quatters** [comment]
> You would need to change source code to support this

Any chances that it will be possible to add new notification methods in the future? I mean without need for changes in the source code.
